### PR TITLE
Reduce ee complexity

### LINKED ISSUE #1117: Reduce complexity in evaluator
The code is very hard to understand and probably the root to the flakyness experienced in the tests

### PR DESCRIPTION
**Pull request by jondequinor**
Resolves #1117 

Reduce evaluator complexity and improve error handling

This is done by
 - logging websocket errors since they include uncaught exceptions
   happening inside the handlers
 - logging for EE now includes thread information
 - inline function definitions are extracted

### CODE REVIEW

File: ert_shared/ensemble_evaluator/context_manager.py [hunk 1]
```diff
@@ -44,6 +45,9 @@ def _attach(run_context, run_path_list, forward_model):
     logger.debug("ee ws started")
 
     event_logs = [Path(path.runpath) / "event_log" for path in run_path_list]
+    for event_log in event_logs:
```
Comment (jondequinor) [inline ert_shared/ensemble_evaluator/context_manager.py:48]: Isn't this file being cleared anyway?

Comment (sondreso) [reply ert_shared/ensemble_evaluator/context_manager.py:48]: Yes, but the problem is that the NSF adaptor reads from the file before the job_dispatch clears it 😅

Comment (jondequinor) [reply ert_shared/ensemble_evaluator/context_manager.py:48]: True. 😂 Well, if these two components (job_dispatch and nfs_adaptor) are to become _more_ concurrent, this strategy will not work. Maybe we should just focus on the `source` of the events, and parse the `ee` bit. If there's a mismatch, then drop the event?

Comment (jondequinor) [reply ert_shared/ensemble_evaluator/context_manager.py:48]: (Some time in the future)

**Review by frode-aarstad — APPROVED** [review]
lgtm

**Pull request merged**